cmd/wc/main.go

```diff
@@ -0,0 +1,70 @@
+package main
+
+import (
+	"bufio"
+	"flag"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+	"unicode/utf8"
+)
+
+func main() {
+	var lPtr, wPtr, cPtr bool
+	flag.BoolVar(&lPtr ,"l", false, "number of lines bool")
+	flag.BoolVar(&wPtr ,"w", false, "number of words bool")
+	flag.BoolVar(&cPtr ,"c", false, "number of characters bool")
+
+	flag.Parse()
+
+	var input io.Reader
+	var fileName string
+	var l int
+	var w int
+	var c int
+
+	idx := flag.NFlag()
+
+	if len(os.Args) == idx+1 {
+		fmt.Println("No file passed")
+		os.Exit(1)
+	}
+
+	fileName = os.Args[idx+1]
+
+	file, err := os.Open(fileName)
+	if err != nil {
+		fmt.Println("Cannot open file!")
+		os.Exit(1)
+	}
+
+	input = file
+
+	scanner := bufio.NewScanner(input)
+	for scanner.Scan() {
+		line := scanner.Text()
+		l++
+
+		words := strings.Fields(scanner.Text())
+		w += len(words)
```

Comment:
why calculate lines, works and chars if the flags are false

Reply:
> why calculate lines, works and chars if the flags are false

saves overhead of having to put conditions here and at the end as well, I can still change it if you want

Reply:
no need for that